021-server-defined-attributes.md

```diff
@@ -0,0 +1,120 @@
+# Optional server-defined attributes
+
+* Creator: Khushboo Verma
+* Relevant Issues: N/A
+
+- [RFC Template](#rfc-template)
+  - [Summary](#summary)
+  - [Resources](#resources)
+  - [Implementation](#implementation)
+    - [API Changes](#api-changes)
+    - [Workers / Commands](#workers--commands)
+    - [Supporting Libraries](#supporting-libraries)
+    - [Data Structures](#data-structures)
+    - [SDKs](#sdks)
+    - [Breaking Changes](#breaking-changes)
+    - [Documentation & Content](#documentation--content)
+  - [Reliability](#reliability)
+    - [Security](#security)
+    - [Scaling](#scaling)
+    - [Benchmarks](#benchmarks)
+    - [Tests (UI, Unit, E2E)](#tests-ui-unit-e2e)
+  - [Open Questions](#open-questions)
+  - [Future Possibilities](#future-possibilities)
+
+## Summary
+<!-- Describe the problem we want to solve and suggested solution in a few paragraphs -->
+This RFC proposes the addition of optional server-defined attributes in all the project collections that can only be added by server and cannot be modified by the client. The user can add them to each collection through the API or console. Some useful sttributes can be `createdBy`, `updatedBy`, `ip` etc.
```

Comment:
Let's have exact list of all internal attributes we will have

Reply:
it would also be good to clarify exactly what the behavior of each would be.